### PR TITLE
Add temperature flag for when generating melodies.

### PR DESCRIPTION
**Pull request by elliotwaite**
Adjusts randomness of the generated melodies by applying:
logits = logits/temperature

### CODE REVIEW

File: magenta/models/attention_rnn/attention_rnn_graph.py [hunk 1]
```diff
@@ -153,6 +153,9 @@ def build_graph(mode, hparams_string, input_size, num_classes,
         tf.add_to_collection('summary_op', summary_op)
 
     elif mode == 'generate':
+      if hparams.temperature != 1.0:
+        logits_flat /= hparams.temperature
```
Comment (adarob) [inline magenta/models/attention_rnn/attention_rnn_graph.py]: any reason to not just always do the division?

**Comment by elliotwaite** [comment]
@adarob, to make the graph more efficient by not adding the division op to the graph when not needed.
